### DISCUSSION TITLE
Polar grid on a flat surface

### DISCUSSION THREAD

**Discussion by fernandorannou** [esmf-org/esmf#258] [ESMPy] [answered]
### Requirements

- [X] Reviewed [ESMF Reference Manual](https://earthsystemmodeling.org/doc/)
- [X] Searched [GitHub Discussions](https://github.com/orgs/esmf-org/discussions?discussions_q=)

### Affiliation(s)

Universidad de Santiago de Chile

### ESMF Version

esmpy

### Issue

Hello,
I'm working on regridding a 2D rectilinear data (log constant in x and y) onto a 2D flat polar grid (log also on r and theta). The documentation says that the polar grid is on a perfect sphere, so I assume I must use a mesh instead of a grid for the destination field, and set to esmpy.MeshElemType.QUAD for elemType  (?)

Can I use a esmpy.CoordSys.CART grid for the source field and a mesh for the destination field?
Thanks 

Fer

### Autotag

@oehmke

**Comment by oehmke** [comment, marked as answer]
Hi Fer,

  You can use an ESMF Grid for the source and a Mesh for the destination. However, the decision about whether to use a Grid or a Mesh is based more on the topology or connections of the geometry and not the coordinates. A Grid only handles logically rectangular (or set of tiles of logically rectangular) geometries. Whereas a Mesh is much more general than that. (What the documentation you refer to means is that when ESMF Grids, Meshes, etc. use spherical coordinates, then they are represented as a perfect sphere within the regrid calculations.)

  If you have a source and destination that are not spherical coordinates, then you can use  esmpy.CoordSys.CART. However, the important thing is that the coordinates are consistent between the two. E.g. If you have a source point and a destination point with the same coordinates, then they should be in the same location. 

  Does that make sense? Let me know if you have more questions. 

Thanks,
- Bob

**Reply by fernandorannou** [reply]
Ok, I'll give it a try. Thanks a lot.
